PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = ['fleet.vehicle.model.category']
-    _inherit = ['stock.location']
-    _inherit = ["stock.picking"]
-    _inherit = ['stock.picking.batch']

Total Changes: 8

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/stock_fleet/models/fleet_vehicle_model.py
+++ b/addons/stock_fleet/models/fleet_vehicle_model.py
@@ -5,7 +5,7 @@ from odoo.tools import format_list
 
 
 class FleetVehicleModelCategory(models.Model):
-    _inherit = ['fleet.vehicle.model.category']
+    _inherit = 'fleet.vehicle.model.category'
 
     weight_capacity = fields.Float(string="Max Weight")
     weight_capacity_uom_name = fields.Char(string='Weight unit of measure label', compute='_compute_weight_capacity_uom_name')

--- a/addons/stock_fleet/models/stock_location.py
+++ b/addons/stock_fleet/models/stock_location.py
@@ -2,6 +2,6 @@ from odoo import fields, models
 
 
 class StockLocation(models.Model):
-    _inherit = ['stock.location']
+    _inherit = 'stock.location'
 
     is_a_dock = fields.Boolean("Is a Dock Location")

--- a/addons/stock_fleet/models/stock_picking.py
+++ b/addons/stock_fleet/models/stock_picking.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class StockPicking(models.Model):
-    _inherit = ["stock.picking"]
+    _inherit = "stock.picking"
 
     zip = fields.Char(related='partner_id.zip', string='Zip', search="_search_zip")
 

--- a/addons/stock_fleet/models/stock_picking_batch.py
+++ b/addons/stock_fleet/models/stock_picking_batch.py
@@ -4,7 +4,7 @@ from odoo import fields, models, api
 
 
 class StockPickingBatch(models.Model):
-    _inherit = ['stock.picking.batch']
+    _inherit = 'stock.picking.batch'
 
     vehicle_id = fields.Many2one('fleet.vehicle', string="Vehicle")
     vehicle_category_id = fields.Many2one(
